source/write/gridfs.txt

```diff
@@ -0,0 +1,225 @@
+collection in the specified bucket and uses the information to reconstruct
+the file from documents in the ``chunks`` collection. You can read the file
+into memory or output it to a stream.
+
+Create a GridFS Bucket
+----------------------
+
+To store or retrieve files from GridFS, create a GridFS bucket by calling the
+``GridFSBucket()`` constructor and passing in a ``MongoDatabase`` instance.
+You can use the ``GridFSBucket`` instance to
+call read and write operations on the files in your bucket.
+
+.. literalinclude:: /includes/write/gridfs.scala
+      :language: scala
+      :dedent:
+      :start-after: start-create-bucket
+      :end-before: end-create-bucket
+
+To create or reference a bucket with a custom name other than the default name
+``fs``, pass your bucket name as the second parameter to the ``GridFSBucket()``
+constructor, as shown in the following example:
+
+.. literalinclude:: /includes/write/gridfs.scala
+      :language: scala
+      :dedent:
+      :start-after: start-create-custom-bucket
+      :end-before: end-create-custom-bucket
+
+Upload Files
+------------
+
+The ``GridFSBucket.uploadFromObservable()`` method reads the contents of an
+``Observable[ByteBuffer]`` and saves it to the ``GridFSBucket`` instance.
+
+You can use the ``GridFSUploadOptions`` type to configure the chunk size or include
+additional metadata.
+
+The following example uploads the contents of an ``Observable[ByteBuffer]`` into
+``GridFSBucket``:
+
+.. literalinclude:: /includes/write/gridfs.scala
+      :language: scala
+      :dedent:
+      :start-after: start-upload-files
+      :end-before: end-upload-files
+
+Retrieve File Information
+-------------------------
+
+In this section, you can learn how to retrieve file metadata stored in the
+``files`` collection of the GridFS bucket. The metadata contains information
+about the file it refers to, including:
+
+- The ``_id`` of the file
+- The name of the file
+- The length/size of the file
+- The upload date and time
+- A ``metadata`` document in which you can store any other information
+
+To learn more about fields you can retrieve from the ``files`` collection, see the
+:manual:`GridFS Files Collection </core/gridfs/#the-files-collection>` documentation in the
+{+mdb-server+} manual.
+
+To retrieve files from a GridFS bucket, call the ``find()`` method on the ``GridFSBucket``
+instance. The following code example retrieves and prints file metadata from all files in
+a GridFS bucket:
+
+.. literalinclude:: /includes/write/gridfs.scala
+      :language: scala
+      :dedent:
+      :start-after: start-retrieve-file-info
+      :end-before: end-retrieve-file-info
+
+To learn more about querying MongoDB, see :ref:`<scala-retrieve>`.
+
+Download Files
+--------------
+
+The ``downloadToObservable()`` method returns an ``Observable[ByteBuffer]``
+that reads the contents from MongoDB.
+
+To download a file by its file ``_id``, pass the ``_id`` to the method.
```

Comment:
```suggestion
To download a file by its file ``_id``, retrieve the ``_id`` and pass is to the method.
```
S: Not 100% necessary, but I think it could make sense to add since we're not showing explicit retrieval of the _id.

Reply:
Users might have the fileID without explicitly retrieving it in code (i.e. looking it up in the Atlas UI) so I'm inclined to leave this specific part as is.